### PR TITLE
Refactor CertUtils. Support ECDSA and PrivateKey.

### PR DESCRIPTION
**Pull request by KostyaSha**
- Fix from #441 without tests.
- My Ecdsa fix that is covered with tests, but they neet to be ported from https://github.com/KostyaSha/yet-another-docker-plugin/tree/master/yet-another-docker-its
- methods more flexible for other use cases
- try-with-resources

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/529)

### CODE REVIEW

**Comment by KostyaSha** [comment]
@marcuslinke could you review?

**Comment by marcuslinke** [comment]
@KostyaSha LGTM :+1:

**Comment by KostyaSha** [comment]
@marcuslinke how do you handle names? I have shell scripts that generates server/client certs, but they should be tied to some hostname/address. I used 192.168.99.100 for my docker-machine but it not ideal.
It should be possible generate certs during container build and then attach volume to DIND + fetch client file from this DND and then try execute connection. Would it be right way?

**Comment by marcuslinke** [comment]
@KostyaSha Currently I manage IP/certs manually in my local `~/.docker-java.properties`. So when IP of docker-machine VM changes I have to modify it. So it looks something like this currently:

```
DOCKER_HOST=tcp://192.168.99.102:2376
DOCKER_CERT_PATH=/Users/marcus/.docker/machine/machines/docker-1.10.2

#DOCKER_HOST=tcp://192.168.99.105:2376
#DOCKER_CERT_PATH=/Users/marcus/.docker/machine/machines/docker-1.8.1

#DOCKER_HOST=tcp://192.168.99.106:2376
#DOCKER_CERT_PATH=/Users/marcus/.docker/machine/machines/docker-1.9.1
```
However when working with DIND we need to generate certs that are tied to the current DOCKER_HOST ip, right? This could be done when starting the DIND container I think. After that we could copy them via https://docs.docker.com/engine/reference/api/docker_remote_api_v1.22/#get-an-archive-of-a-filesystem-resource-in-a-container to our local java env. WDYT?

**Comment by KostyaSha** [comment]
First of all i found that openssl tool may differ, so we should run generation in reproducible env. 
Host would be hostname from outer dockerClient connection (i like `@JunitRule`/`@ExternalResource` instead of abstract subclasses).

dind easily runs, but i don't remember about port in cert validability... I could tre reimplement my tests.

**Comment by KostyaSha** [comment]
And this wouldn't work because netty uses sslcontext now?

**Comment by KostyaSha** [comment]
Found, it wrapped now.. Maybe let's merge without tests? Too much time left...

File: src/main/java/com/github/dockerjava/core/util/CertificateUtils.java [hunk 1]
```diff
@@ -41,115 +49,157 @@ public static boolean verifyCertificatesExist(String dockerCertPath) {
...
     }
 
+    /**
+     * @param dockerCertPath with standard named files.
+     */
     public static KeyStore createKeyStore(final String dockerCertPath) throws NoSuchAlgorithmException,
             InvalidKeySpecException, IOException, CertificateException, KeyStoreException {
-        KeyPair keyPair = loadPrivateKey(dockerCertPath);
-        List<Certificate> privateCertificates = loadCertificates(dockerCertPath);
+        return createKeyStore("key.pem", "cert.pem");
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/util/CertificateUtils.java]: mm...

Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/core/util/CertificateUtils.java]: Trying to fix locally

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/util/CertificateUtils.java]: Smells like i missed this wrapper when refactored methods to be reusable for non-file based usage.
Could you fix or should i?

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/util/CertificateUtils.java]: i guess it should just load files into string, or pass streams and everything would work

Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/core/util/CertificateUtils.java]: Tried to load into strings but I ended with a `javax.ws.rs.ProcessingException: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path validation failed: java.security.cert.CertPathValidatorException: signature check failed 
`

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/util/CertificateUtils.java]: maybe ca.cert is missing?

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/util/CertificateUtils.java]: call to createTrustStore...

Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/core/util/CertificateUtils.java]: Right, had a typo there. Now it seems to work.

**Pull request merged**